river/ensemble/streaming_random_patches.py

```diff
@@ -568,3 +353,690 @@ def random_subspace(all_features: list, k: int, rng: np.random.Generator):
+    simulates bagging or random subspaces. The default algorithm uses both
+    bagging and random subspaces, namely Random Patches. The default base
+    estimator is a Hoeffding Tree, but other base estimators can be used
+    (differently from random forest variations).
+
+    Parameters
+    ----------
+    model
+        The base estimator.
+    n_models
+        Number of members in the ensemble.
+    subspace_size
+        Number of features per subset for each classifier where `M` is the
+        total number of features.<br/>
+        A negative value means `M - subspace_size`.<br/>
+        Only applies when using random subspaces or random patches.<br/>
+        * If `int` indicates the number of features to use. Valid range [2, M]. <br/>
+        * If `float` indicates the percentage of features to use, Valid range (0., 1.]. <br/>
+        * 'sqrt' - `sqrt(M)+1`<br/>
+        * 'rmsqrt' - Residual from `M-(sqrt(M)+1)`
+    training_method
+        The training method to use.<br/>
+        * 'subspaces' - Random subspaces.<br/>
+        * 'resampling' - Resampling.<br/>
+        * 'patches' - Random patches.
+    lam
+        Lambda value for resampling.
+    drift_detector
+        Drift detector.
+    warning_detector
+        Warning detector.
+    disable_detector
+        Option to disable drift detectors:<br/>
+        * If `'off'`, detectors are enabled.<br/>
+        * If `'drift'`, disables concept drift detection and the background learner.<br/>
+        * If `'warning'`, disables the background learner and ensemble members are
+         reset if drift is detected.<br/>
+    disable_weighted_vote
+        If True, disables weighted voting.
+    seed
+        Random number generator seed for reproducibility.
+    metric
+        Metric to track members performance within the ensemble.
+
+    Examples
+    --------
+    >>> from river import synth
+    >>> from river import ensemble
+    >>> from river import tree
+    >>> from river import evaluate
+    >>> from river import metrics
+
+    >>> dataset = synth.ConceptDriftStream(seed=42, position=500,
+    ...                                    width=50).take(1000)
+    >>> base_model = tree.HoeffdingTreeClassifier(
+    ...     grace_period=50, split_confidence=0.01,
+    ...     nominal_attributes=['age', 'car', 'zipcode']
+    ... )
+    >>> model = ensemble.SRPClassifier(
+    ...     model=base_model, n_models=3, seed=42,
+    ... )
+    >>> metric = metrics.Accuracy()
+
+    >>> evaluate.progressive_val_score(dataset, model, metric)  # doctest: +SKIP
+    Accuracy: 74.47%
+
+    References
+    ----------
+    [^1]: Heitor Murilo Gomes, Jesse Read, Albert Bifet.
+          Streaming Random Patches for Evolving Data Stream Classification.
+          IEEE International Conference on Data Mining (ICDM), 2019.
+
+    """
+
+    def __init__(
+        self,
+        model: base.Estimator = None,
+        n_models: int = 100,
```

Comment:
is this correct? (the default number of base learners)

A high number of learners will consequently make the learning process slower. Maybe novice users could be discouraged from using SRP due to runtime. Am I overthinking?

Reply:
It is the number specified in the original implementation, but I agree that it is quite a concern. Should we change it and add a note?